gpumon/gpumon_client.ml

```diff
@@ -17,14 +17,13 @@ open Xcp_client
 
 let xml_url () = "file:" ^ xml_path
 
-module Client = Gpumon_interface.Client(struct
-	let rpc call =
-		if !use_switch
-		then json_switch_rpc queue_name call
-		else xml_http_rpc
-                        ~srcstr:(get_user_agent ())
-                        ~dststr:"gpumon"
-                        xml_url
-                        call
-end)
-
+let rpc call =
+  if !use_switch
+  then json_switch_rpc queue_name call
+  else xml_http_rpc
+      ~srcstr:(get_user_agent ())
+      ~dststr:"gpumon"
+      xml_url
+      call
+module Client = RPC_API(Idl.GenClientExnRpc(struct let rpc=rpc end))
+include Client
```

Comment:
Why did you decide to include the client? Just to avoid doing `Gpumon_client.Client`?

Reply:
Yes, although it looks like that's redundant since it's only called in xapi 4 times and always as `Gpumon_client.Client` anyway, so I can change that.

Reply:
It's fine either way